### PR TITLE
Code health - Convert exptest to new framework

### PR DESCRIPTION
**Pull request by jon-oracle**
- [x] tests are added or updated

Converted modular exponentiation tests to new framework. Standardized the test output.

### CODE REVIEW

File: test/exptest.c [hunk 1]
```diff
@@ -18,10 +18,18 @@
...
+#include "test_main.h"
+#include "testutil.h"
+
 #define NUM_BITS        (BN_BITS2 * 4)
 
-static const char rnd_seed[] =
-    "string to make the random number generator think it has entropy";
+#define BN_print_var(v) bn_print_var(#v, v)
+
+static void bn_print_var(const char *var, const BIGNUM *bn) {
```
Comment (richsalz) [inline test/exptest.c]: fix culry brace.

File: test/exptest.c [hunk 2]
```diff
@@ -54,76 +59,68 @@ static int test_exp_mod_zero()
...
-
     p = BN_new();
-    if (!p)
-        goto err;
-    BN_zero(p);
-
     r = BN_new();
-    if (!r)
+
+    if (!TEST_ptr(m)
```
Comment (richsalz) [inline test/exptest.c]: I got convinced to put the assignments inside the TESTptr lines...

File: test/exptest.c [hunk 3]
```diff
@@ -134,112 +131,84 @@ static int test_exp_mod_zero()
...
+    BIGNUM *r_recp = NULL;
+    BIGNUM *r_simple = NULL;
+    BIGNUM *a = NULL;
+    BIGNUM *b = NULL;
+    BIGNUM *m = NULL;
 
     ctx = BN_CTX_new();
-    if (ctx == NULL)
-        EXIT(1);
+    if (!TEST_ptr(ctx))
```
Comment (richsalz) [inline test/exptest.c]: again, put all the assignments inside the TEST_ptr calls.

File: test/build.info [hunk 4]
```diff
@@ -139,7 +139,7 @@ IF[{- !$disabled{tests} -}]
   INCLUDE[dsatest]=.. ../include
   DEPEND[dsatest]=../libcrypto
 
-  SOURCE[exptest]=exptest.c
+  SOURCE[exptest]=exptest.c testutil.c test_main.c
```
Comment (levitte) [inline test/build.info]: Remove that change...

File: test/build.info [hunk 5]
```diff
@@ -139,7 +139,7 @@ IF[{- !$disabled{tests} -}]
...
   INCLUDE[exptest]=../include
   DEPEND[exptest]=../libcrypto
```
Comment (levitte) [inline test/build.info]: ... and add ` libtestutil.a` here instead.

File: test/exptest.c [hunk 6]
```diff
@@ -18,112 +18,106 @@
 #include <openssl/rand.h>
 #include <openssl/err.h>
 
+#include "test_main.h"
```
Comment (levitte) [inline test/exptest.c]: Remove this include.

**Comment by levitte** [comment]
Also, I'd have an extra look that all lines fit within 80 columns.

**Comment by jon-oracle** [comment]
I've rebased and updated the test. compiles and runs now.

**Comment by mattcaswell** [comment]
Needs merge

**Pull request closed without merging**